src/instructions/mod.rs

```diff
@@ -22,10 +20,54 @@ fn sign_extend(mut x: u16, bit_count: u8) -> u16 {
     x
 }
 
+pub enum Opcode {
+    OpBr,   // Branch
+    OpAdd,  // Add
+    OpLd,   // Load
+    OpSt,   // Store
+    OpJsr,  // Jump to Subroutine
+    OpAnd,  // Bitwise AND
+    OpLdr,  // Load Base+Offset
+    OpStr,  // Store Base+Offset
+    OpRti,  // Return from Interrupt (Unused)
+    OpNot,  // Bitwise NOT
+    OpLdi,  // Load Indirect
+    OpSti,  // Store Indirect
+    OpJmp,  // Jump
+    OpRes,  // Reserved (Unused)
+    OpLea,  // Load Effective Address
+    OpTrap, // Trap
+}
+
+/// Convert u16 into Opcode
+impl TryFrom<u16> for Opcode {
+    type Error = ();
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Opcode::OpBr),
+            1 => Ok(Opcode::OpAdd),
+            2 => Ok(Opcode::OpLd),
+            3 => Ok(Opcode::OpSt),
+            4 => Ok(Opcode::OpJsr),
+            5 => Ok(Opcode::OpAnd),
+            6 => Ok(Opcode::OpLdr),
+            7 => Ok(Opcode::OpStr),
+            8 => Ok(Opcode::OpRti),
+            9 => Ok(Opcode::OpNot),
+            10 => Ok(Opcode::OpLdi),
+            11 => Ok(Opcode::OpSti),
+            12 => Ok(Opcode::OpJmp),
+            13 => Ok(Opcode::OpRes),
+            14 => Ok(Opcode::OpLea),
+            15 => Ok(Opcode::OpTrap),
+            _ => Err(()),
+        }
```

Comment:
Maybe you could leave in each opcode function a little explanation of what the operation does